src/res_comments_table.ml

```diff
@@ -1088,7 +1088,9 @@ and walkExpression expr t comments =
       walkExpression expr2 t inside;
       attach t.trailing expr2.pexp_loc trailing
   | Pexp_ifthenelse (ifExpr, thenExpr, elseExpr) -> (
-    let leading, inside, trailing = partitionByLoc comments ifExpr.pexp_loc in
+    let leading, rest = partitionLeadingTrailing comments expr.pexp_loc in
+    attach t.leading expr.pexp_loc leading;
+    let leading, inside, trailing = partitionByLoc rest ifExpr.pexp_loc in
```

Comment:
This logic makes perfect sense.
Question: why is this not the logic applied to every node?
(except value and type def for example)

Reply:
E.g. isn't `Pexp_setfield` suffering from the same issue?

Reply:
I don't know really - do you mean doing this at the beginning of `walkExpression` before the match on `pexp_desc`?
Maybe @IwanKaramazow can comment.

Reply:
Don't even know if this is only for expressions.

Reply:
I can't remember, the comment table mostly evolved in an ad-hoc way… It does make sense to apply this logic to every node.